variables.tf

```diff
@@ -1,9 +1,67 @@
+##############################################################################
+# Common variables
+##############################################################################
+
+variable "resource_group_id" {
+  type        = string
+  description = "The resource group ID where resources will be provisioned."
+}
+
+variable "region" {
+  description = "The region to provision the resources."
+  type        = string
+  default     = "us-south"
+
+  validation {
+    condition     = contains(["us-east", "us-south", "eu-gb", "au-syd"], var.region)
+    error_message = "Value for region must be one of the following: ${join(", ", ["us-east", "us-south", "eu-gb", "au-syd"])}"
+  }
+}
+
 ########################################################################################################################
-# Input Variables
+# App Config Instance Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "app_config_name" {
+  type        = string
+  description = "Name for the App Configuration service instance"
+}
+
+variable "app_config_plan" {
+  type        = string
+  description = "Plan for the App Configuration service instance"
+  default     = "standardv2"
+
+  validation {
+    condition     = contains(["lite", "standardv2", "enterprise"], var.app_config_plan)
+    error_message = "Value for plan must be one of the following: \"lite\", \"standard\", or \"enterprise\"."
+  }
+}
+
+variable "app_config_service_endpoints" {
+  type        = string
+  description = "Service Endpoints for the App Configuration service instance"
+  default     = "public"
+
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.app_config_service_endpoints)
```

Comment:
Does it support private only? Did you try it? I see this in the UI:
![image](https://github.com/terraform-ibm-modules/terraform-ibm-app-configuration/assets/29608224/5b8e567c-697a-4681-8f69-8b1b3e9f84e0)

Reply:
I misread docs, based on https://cloud.ibm.com/docs/app-configuration?topic=app-configuration-ac-create-an-instance this only support both and even then it's only for enterprise plans, fixing this var and adding validation in the module to check it's enterprise if both are enabled.